### PR TITLE
feat(cli): addr-conversion

### PR DESCRIPTION
**Pull request by rootulp**
## Motivation

We got a question on Slack on how to convert from a celestia1xxx to a celestiavaloper1xxx address. The implementation in this PR was inspired by [zeta-chain](https://github.com/zeta-chain/node/blob/fca927f538cf4cd89e74db57a262f7b8ab5a91af/cmd/zetacored/addr_converter.go#L11)

## Testing

```shell
$ ./build/celestia-appd addr-conversion --help
Reads a celestia1xxx or celestiavaloper1xxx address and converts it to the other type.

Usage:
  celestia-appd addr-conversion [celestia address] [flags]

Examples:
celestia-appd addr-conversion celestia1grvklux2yjsln7ztk6slv538396qatckqhs86z
celestia-appd addr-conversion celestiavaloper1grvklux2yjsln7ztk6slv538396qatck9gj7vy

$ celestia-appd addr-conversion celestia1grvklux2yjsln7ztk6slv538396qatckqhs86z
celestiavaloper1grvklux2yjsln7ztk6slv538396qatck9gj7vy

$ celestia-appd addr-conversion celestiavaloper1grvklux2yjsln7ztk6slv538396qatck9gj7vy
celestia1grvklux2yjsln7ztk6slv538396qatckqhs86z
```

### CODE REVIEW

**Comment by rootulp** [comment]
I can improve the tests

File: cmd/celestia-appd/cmd/addr_converter.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+func Convert(original string) (string, error) {
+	acc, err := sdk.AccAddressFromBech32(original)
+	if err == nil {
+		return sdk.ValAddress(acc.Bytes()).String(), nil
+	}
+	valAddr, err := sdk.ValAddressFromBech32(original)
+	if err == nil {
+		return sdk.AccAddress(valAddr.Bytes()).String(), nil
+	}
```
Comment (evan-forbes) [inline cmd/celestia-appd/cmd/addr_converter.go:43]: clever error handling, instant approve

**Review by evan-forbes — APPROVED** [review]

File: cmd/celestia-appd/cmd/addr_converter_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		output, err := executeCmd(cmd.AddrConversionCmd(), accAddr)
+		assert.NoError(t, err)
+		assert.Equal(t, valAddr+"\n", output)
+	})
+	t.Run("converts a valoper address", func(t *testing.T) {
+		output, err := executeCmd(cmd.AddrConversionCmd(), valAddr)
+		assert.NoError(t, err)
+		assert.Equal(t, accAddr+"\n", output)
+	})
+	t.Run("returns an error for an invalid address", func(t *testing.T) {
```
Comment (staheri14) [inline cmd/celestia-appd/cmd/addr_converter_test.go:25]: Everything has been thoroughly addressed in the tests 👍. One final enhancement we could make is to include a test for scenarios where an invalid valoper address is provided.

**Review by staheri14 — DISMISSED** [review]
Looks good! Left some comments, all non-blocking.

**Comment by the-smooth-operator** [comment]
I've tested this and works like a charm, thank you! This is going to be very helpful for us

File: cmd/celestia-appd/cmd/root.go [hunk 3]
```diff
@@ -140,6 +140,7 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig encoding.Config) {
 		commands.CompactGoLevelDBCmd,
 		addrbookCommand(),
 		downloadGenesisCommand(),
+		AddrConversionCmd(),
```
Comment (cmwaters) [inline cmd/celestia-appd/cmd/root.go:143]: Does it make more sense to put this under the debug sub command where the other conversion like commands are based.

File: cmd/celestia-appd/cmd/addr_converter.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+)
+
+// AddrConversionCmd returns a command that converts between celestia1xxx and
+// celestiavaloper1xxx addresses.
+func AddrConversionCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "addr-conversion [celestia address]",
+		Short: "Convert a celestia1xxx address to a validator operator address celestiavaloper1xxx",
```
Comment (cmwaters) [inline cmd/celestia-appd/cmd/addr_converter.go:15]: What about to the consensus address (celestiavalcons)? Is that necessary?

See https://docs.cosmos.network/main/build/spec/addresses/bech32#hrp-table

Comment (cmwaters) [reply cmd/celestia-appd/cmd/addr_converter.go:15]: I would like to think there's some existing tooling here we can import from the SDK then having to write it ourselves but maybe I'm wrong

File: cmd/celestia-appd/cmd/root.go (hunk 3 above)
Comment (cmwaters) [reply cmd/celestia-appd/cmd/root.go:143]: Don't worry if this is too difficult to do

**Comment by rootulp** [comment]
Thanks for all the feedback! I agree with all of it. 

I'm not going to work on this PR further b/c I think @the-smooth-operator was unblocked by using it already. If we get another request for a similar CLI command, I'll revive this PR and address the feedback.

**Pull request closed without merging**